internal/dockercompose/compose.go

```diff
@@ -2,55 +2,150 @@ package dockercompose
+	if df == "" && svcConfig.Build.Context != "" {
+		// We only expect a Dockerfile if there's a build context specified.
+		df = "Dockerfile"
 	}
 
-	args = append(args, "--services")
-	servicesText, err := dcOutput(ctx, args...)
+	return Service{
+		Name:    name,
+		Context: svcConfig.Build.Context,
+		DfPath:  path.Join(svcConfig.Build.Context, df),
+	}, nil
+}
+
+func svcNames(ctx context.Context, configPath string) ([]string, error) {
+	servicesText, err := dcOutput(ctx, configPath, "config", "--services")
 	if err != nil {
-		return nil, files, err
+		return nil, err
 	}
 
 	serviceNames := strings.Split(string(servicesText), "\n")
 
-	var services []Service
+	var result []string
 
 	for _, name := range serviceNames {
 		if name == "" {
 			continue
 		}
-		services = append(services, Service{Name: name})
+		result = append(result, name)
+	}
+
+	return result, nil
+}
+
+func ParseConfig(ctx context.Context, configPath string) ([]Service, error) {
+	configOut, err := dcOutput(ctx, configPath, "config")
+	if err != nil {
+		return nil, err
+	}
+
+	config := Config{}
+	err = yaml.Unmarshal([]byte(configOut), &config)
+	if err != nil {
+		return nil, err
+	}
+
+	svcNames, err := svcNames(ctx, configPath)
+	if err != nil {
+		return nil, err
+	}
+
+	var services []Service
+
+	for _, name := range svcNames {
+		if name == "" {
+			continue
+		}
+		svc, err := config.GetService(name)
+		if err != nil {
+			return nil, errors.Wrapf(err, "getting service %s", name)
+		}
+		services = append(services, svc)
+	}
+
+	return services, nil
+}
+
+func (s Service) ToManifest(dcConfigPath string) (model.Manifest, error) {
+	m := model.Manifest{
+		Name:       model.ManifestName(s.Name),
+		DcYAMLPath: dcConfigPath,
+	}
+
+	if s.DfPath == "" {
+		// DC service may not have Dockerfile -- e.g. may be just an image that we pull and run.
+		return m, nil
+	}
+
+	// TODO(maia): record s.DfPath as config file!
+	bs, err := ioutil.ReadFile(s.DfPath)
+	if err != nil {
+		return model.Manifest{}, errors.Wrapf(err, "opening Dockerfile for %s: '%s'", s.Name, s.DfPath)
+	}
+
+	// TODO(maia): ignore volumes mounted via dc.yml (b/c those auto-update)
+	df := dockerfile.Dockerfile(bs)
+	mounts, err := df.DeriveMounts(s.Context)
+	if err != nil {
+		return model.Manifest{}, err
 	}
 
-	return services, files, nil
+	m.Mounts = mounts
+	return m, nil
 }
 
-func dcOutput(ctx context.Context, args ...string) (string, error) {
+func dcOutput(ctx context.Context, configPath string, args ...string) (string, error) {
+	args = append([]string{"-f", configPath}, args...)
 	output, err := exec.CommandContext(ctx, "docker-compose", args...).Output()
 	if err != nil {
 		errorMessage := fmt.Sprintf("command 'docker-compose %q' failed.\nerror: '%v'\nstdout: '%v'", args, err, string(output))
 		if err, ok := err.(*exec.ExitError); ok {
 			errorMessage += fmt.Sprintf("\nstderr: '%v'", string(err.Stderr))
 		}
-		err = errors.New(errorMessage)
+		err = fmt.Errorf(errorMessage)
```

Comment:
are you sure you want to make this change? note that this will crash if errorMessage contains any format specifiers. errors.New seemed like the right function?